File: 美团8.16笔试/t1.py
import sys

LENGTH = 30
UPPER = (1 << LENGTH) - 1

def len_of_bits(x):
    ans: int = 0
    while x:
        ans += 1
        x >>= 1
    if not ans:
        ans = 1
    return ans

def main():
    input = sys.stdin.read().strip('\r\n\t ').split()
    it = iter(input)
    T = int(next(it))
    ans = []
    for _ in range(T):
        n = int(next(it))
        arr = [int(next(it)) for _ in range(n)]
        s = sum(arr)
        _max = max(arr)
        len = len_of_bits(_max)
        mask = (1 << len) - 1
        upper = UPPER
        for ai in arr:
            upper &= ai
        upper &= mask
        x: int = mask ^ upper # 通过和mask异或，来翻转0和1
        s += x
        ans.append(f"{s} {x}")
    print("\n".join(ans))

if __name__ == "__main__":
    main()

File: 美团8.16笔试/t1.cpp

#include <algorithm>
#include <iostream>
#include <cstdio>

constexpr int N = 505, LEN = 30;
constexpr int UPPER = (1 << LEN) - 1;
int arr[N], T;

int len_of_bits(int x) {
    int ans = 0;
    while (x) {
        ans++;
        x>>=1;
    }
    return ans?ans:1;
}

int main() {
    freopen("input.txt", "r", stdin);
    scanf("%d", &T);
    int len;
    while (T--) {
        scanf("%d", &len);
        int _max = -1, sum = 0;
        for (int i = 1; i <= len; i++) {
            scanf("%d", &arr[i]);
            sum += arr[i];
            _max = std::max(_max, arr[i]);
        }
        int len = len_of_bits(_max);
        int upper = UPPER, mask = (1 << len) - 1;
        for (int i = 1; i <= len; i++) {
            upper &= arr[i];
        }
        upper &= mask;
        int x = mask ^ upper;
        sum += x;
        printf("%d %d\n", sum, x);
    }
    return 0;
}